odc/stats/tasks.py

```diff
@@ -151,18 +158,65 @@ def match_dss(groups, group_size):
         )
         grouped_dss = match_dss(grouped_dss, group_size)
 
-        def map_fuse_func(x):
-            return fuse_ds(*x, product=product)
+        if fuse_dss:
+            fused_product = fuse_products(*products)
+
+            def map_fuse_func(x):
+                return fuse_ds(*x, product=fused_product)
+
+            dss = map(map_fuse_func, grouped_dss)
+            return dss
+        else:
+            return grouped_dss
+
+    @classmethod
+    def _find_dss(
+        cls,
+        dc: Datacube,
+        products: str,
+        query: Dict[str, Any],
+        dataset_filter=None,
+        predicate=None,
+    ):
+        if dataset_filter is None:
+            dataset_filter = {}
+
+        # query by a list of products is not a "officially" supported feature
+        # but it is embedded in the code everywhere
+        # mark it for ref
+
+        product_list = re.split(r"\+|-", products)
+        product_list = list(filter(None, product_list))
+        group_size = len(re.findall(r"\+", products))
+
```

Comment:
I don't understand what's going on here.  What is the expected format of "products"?  That's some weird stuff going on with pluses and minuses.  

`products="prod1+prod2-prod3+prod4"` and `products="+++prod1+prod2-+-prod3++prod4"` both  yield the same 4 member `product_list`, but one givess a `group_size` of 4 and the other 7??  With no warning/error message?

It seems like the main purpose of `find_dss` is to expand a product string into a product list, so a little bit of documentation here would be helpful.

(It's still better than it was, mind you.)